fhir-flow/src/main/java/com/ibm/fhir/flow/api/IFlowInteractionHandler.java

```diff
@@ -15,17 +15,17 @@ public interface IFlowInteractionHandler {
 
     /**
      * Perform a logical DELETE interaction
-     * @param changeId
+     * @param entryId
      * @param identifier
      */
-    void delete(long changeId, ResourceIdentifier identifier);
+    void delete(String entryId, ResourceIdentifier identifier);
 
     /**
      * Perform a create-or-update (PUT) interaction
-     * @param changeId
+     * @param entryId
      * @param identifier
      * @param resourceData
      * @param resource
      */
-    void createOrUpdate(long changeId, ResourceIdentifier identifier, String resourceData, Resource resource);
+    void createOrUpdate(String entryId, ResourceIdentifier identifier, String resourceData, Resource resource);
```

Comment:
will it work if entryId is null?

Reply:
I've added handling in the Bundle processing so we create our own value if the entry.id value is null. Only used for logging to assist in correlating messages.